test/helpers/gem_helpers.rb

```diff
@@ -17,7 +17,7 @@ def build_gem(name, version, summary = "Gemcutter", platform = "ruby", &block)
     build_gemspec(new_gemspec(name, version, summary, platform, &block))
   end
 
-  def new_gemspec(name, version, summary, platform, &block)
+  def new_gemspec(name, version, summary, platform, metadata = nil, &block)
```

Comment:
@bf4 @simi As stated [before][], A parsed gemspec can return a nil `metadata` hash. This is in `Gem::Package.new("filename").spec` and not part of the rubygems code base.

Changing `linkset.rb` and `new_gemspec` will pass for tests using the `new_gemspec` factory, but blows up when parsing real gems.

[before]: https://github.com/rubygems/rubygems.org/pull/724#discussion_r19806413